.github/workflows/build.yml

```diff
@@ -24,34 +24,61 @@ on:
     types: [opened, synchronize, reopened]
 jobs:
   build:
-    name: Build and analyze
+    name: Maven Build
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3.3.0
-        with:
-          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+      - name: Git clone
+        uses: actions/checkout@v4
       - name: Set up JDK 11
-        uses: actions/setup-java@v3.10.0
+        uses: actions/setup-java@v4
         with:
           java-version: 11
           distribution: temurin
-      - name: Cache SonarCloud packages
-        uses: actions/cache@v3.2.6
-        with:
-          path: ~/.sonar/cache
-          key: ${{ runner.os }}-sonar
-          restore-keys: ${{ runner.os }}-sonar
       - name: Cache Maven packages
-        uses: actions/cache@v3.2.6
+        uses: actions/cache@v4
         with:
           path: ~/.m2
           key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
           restore-keys: ${{ runner.os }}-m2
-      - name: Build and analyze
+      - name: Build
+        run: mvn -B verify -Pcoverage
```

Comment:
In order to fully replace ASF Jenkins we probably need to add profile `integrationTesting` (https://github.com/apache/jackrabbit-oak/blob/4a03d67dffd5619c41f5b54ccacfa49e1eb57daf/Jenkinsfile#L29C5-L43C6), however I would refrain from building each module individually but rather build the whole reactor with that profile.

Also `-DtrimStackTrace=false -Dnsfixtures=SEGMENT_TAR,DOCUMENT_NS -Dmongo.db=MongoMKDB-${MONGODB_SUFFIX}` would need to be added, right?